mock_drivers/rosbag_mock_drivers/src/USFControllerDriver.cpp

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright (C) 2020-2021 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+//head files required by USF controller driver
+#include <ros/ros.h>
+#include "rosbag_mock_drivers/VehicleCmd.h"
+
+#include <iostream>
+#include <string>
+#include <cstdio>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <pthread.h>
+//head files required by USF controller driver
+
+#include "rosbag_mock_drivers/MockControllerDriver.h"
+
+namespace mock_drivers
+{
+std::vector<DriverType> MockControllerDriver::getDriverTypes()
+{
+  return { DriverType::CONTROLLER };
+}
+
+uint8_t MockControllerDriver::getDriverStatus()
+{
+  return cav_msgs::DriverStatus::OPERATIONAL;
+}
+
+//have a new vehicle command callback function below
+/*void MockControllerDriver::vehicleCmdCallback(const autoware_msgs::VehicleCmd::ConstPtr& msg)
+{
+  robot_enabled_ = true;  // If a command was received set the robot enabled status to true
+}*/
+
+bool MockControllerDriver::enableRoboticSrv(const cav_srvs::SetEnableRobotic::Request& req,
+                                            cav_srvs::SetEnableRobotic::Response& res)
+{
+  if (robot_enabled_ && req.set == cav_srvs::SetEnableRobotic::Request::ENABLE)
+  {
+    robot_active_ = true;
+  }
+  else
+  {
+    robot_active_ = false;
+  }
+
+  return true;
+}
+
+MockControllerDriver::MockControllerDriver(bool dummy)
+{
+  mock_driver_node_ = MockDriverNode(dummy);
+
+  robot_status_ptr_ =
+      boost::make_shared<ROSComms<cav_msgs::RobotEnabled>>(CommTypes::pub, false, 10, robot_status_topic_);
+  
+  ////vehicle command topic is subscribed by the vehiclesender node.
+
+  /*vehicle_cmd_ptr_ = boost::make_shared<ConstPtrRefROSComms<autoware_msgs::VehicleCmd>>(
+      std::bind(&MockControllerDriver::vehicleCmdCallback, this, std::placeholders::_1), CommTypes::sub, false, 10,
+      vehicle_cmd_topic_);*/
+
```

Comment:
There needs to also be a "vehicle/twist" topic (geometry_msgs/TwistStamped) published which exposes the current vehicle speed.